tests/Struct/FindologicProductTest.php

```diff
@@ -1210,4 +1212,44 @@ private function translateBooleanValue(bool $value): string
 
         return $this->getContainer()->get('translator')->trans($translationKey);
     }
+
+    public function testProductListPrice(): void
+    {
+        $productEntity = $this->createTestProduct(
+            [
+                'price' => [
+                    [
+                        'currencyId' => Defaults::CURRENCY,
+                        'gross' => 50,
+                        'net' => 40,
+                        'linked' => false,
+                        'listPrice' => [
+                            'net' => 20,
+                            'gross' => 25,
+                            'linked' => false
+                        ]
+                    ]
+                ],
+            ]
+        );
+
+        $findologicProductFactory = new FindologicProductFactory();
+        $findologicProduct = $findologicProductFactory->buildInstance(
+            $productEntity,
+            $this->router,
+            $this->getContainer(),
+            $this->shopkey,
+            [],
+            new XMLItem('123')
+        );
+
+        foreach ($findologicProduct->getProperties() as $property) {
```

Comment:
Does this test fail in case both `old_price` and `old_price_net` are not set?

Reply:
No I dont think so. Will make the necessary changes.